### PR TITLE
Couple of view implementations

### LINKED ISSUE #21: Show a debt-creation-View.

### LINKED ISSUE #22: Create a template for input of detailed info of debt.

### LINKED ISSUE #26: Show a debt-settlement-view.

### LINKED ISSUE #28: A debt cannot be settle for more than the amount owed.

### PR DESCRIPTION
**Pull request by gabrielbrattgard**
Implemented
- AddDebtView
- UserCardView
- PickDebtAdapter (untested)

closes #26, closes #21, closes #28, closes #22

### CODE REVIEW

File: app/src/main/java/com/goayo/debtify/view/AddDebtFragment.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import com.goayo.debtify.databinding.PickUsersFragmentBinding;
+import com.goayo.debtify.model.ModelEngine;
+import com.goayo.debtify.modelaccess.IUserData;
+import com.goayo.debtify.view.adapter.PickUserAdapter;
+import com.goayo.debtify.view.adapter.UserCardViewAdapter;
+
+import java.util.Set;
+
+/**
+ * @author Alex Phu, Yenan Wang
```
Comment (Puh00) [inline app/src/main/java/com/goayo/debtify/view/AddDebtFragment.java:27]: Author should be @gabrielbrattgard name? Not my name.

File: app/src/main/java/com/goayo/debtify/view/AddDebtFragment.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+public class AddDebtFragment extends Fragment {
+
+    private AddDebtFragmentBinding binding;
+    private ModelEngine model;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        binding = DataBindingUtil.inflate(inflater, R.layout.add_debt_fragment, container, false);
+        model = ModelEngine.getInstance();
```
Comment (Puh00) [inline app/src/main/java/com/goayo/debtify/view/AddDebtFragment.java:44]: Communicate to the model through ViewModel.

File: app/src/main/java/com/goayo/debtify/view/SettleDebtFragment.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ * @date 2020-09-09
+ * <p>
+ * Page for paying off debt.
+ *
+ * 2020-09-18 Modified by Yenan & Gabriel: Added SettleDebt view, tested with hard-coded value
+ */
+public class SettleDebtFragment extends Fragment {
+
+    private SettleDebtFragmentBinding binding;
+    private ModelEngine model;
```
Comment (OscarSanner) [inline app/src/main/java/com/goayo/debtify/view/SettleDebtFragment.java:37]: This has to be changed. No imports from model. Change to a view model.

Comment (OscarSanner) [reply app/src/main/java/com/goayo/debtify/view/SettleDebtFragment.java:37]: Will be changed when creating of viewModel.

File: app/src/main/java/com/goayo/debtify/view/AddDebtFragment.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    private void initRecyclerView(Set<IUserData> userData) {
+        RecyclerView borrowerRecyclerView = binding.borrowerRecyclerView;
+        RecyclerView lenderRecyclerView = binding.lenderRecyclerView;
+
+        UserCardViewAdapter borrowerAdapter = new UserCardViewAdapter("Select borrower", "", new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                openPickUser();
+            }
+        });
```
Comment (Puh00) [inline app/src/main/java/com/goayo/debtify/view/AddDebtFragment.java:59]: Incredible solution!

File: app/src/main/java/com/goayo/debtify/view/SettleDebtFragment.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import com.goayo.debtify.R;
+import com.goayo.debtify.databinding.SettleDebtFragmentBinding;
+import com.goayo.debtify.model.ModelEngine;
+import com.goayo.debtify.modelaccess.IDebtData;
+import com.goayo.debtify.view.adapter.PickDebtAdapter;
+
+import java.util.List;
+
+/**
+ * @author Alex Phu, Yenan Wang
```
Comment (Puh00) [inline app/src/main/java/com/goayo/debtify/view/SettleDebtFragment.java]: Change to `@author Gabriel Brattgård, Yenan Wang`

File: app/src/main/java/com/goayo/debtify/view/SettleDebtFragment.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+public class SettleDebtFragment extends Fragment {
+
+    private SettleDebtFragmentBinding binding;
+    private ModelEngine model;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        binding = DataBindingUtil.inflate(inflater, R.layout.settle_debt_fragment, container, false);
+        model = ModelEngine.getInstance();
```
Comment (Puh00) [inline app/src/main/java/com/goayo/debtify/view/SettleDebtFragment.java:43]: Communicate through custom ViewModel instead.

File: app/src/main/java/com/goayo/debtify/view/AddDebtFragment.java (hunk 1 above)
Comment (gabrielbrattgard) [reply app/src/main/java/com/goayo/debtify/view/AddDebtFragment.java:27]: Done, changed it and added correct header comment.

File: app/src/main/java/com/goayo/debtify/view/adapter/PickDebtAdapter.java [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+            debtRadioButton = itemView.findViewById(R.id.debtRadioButton);
+            date = itemView.findViewById(R.id.debtDate);
+            lender = itemView.findViewById(R.id.debtLender);
+            borrower = itemView.findViewById(R.id.debtBorrower);
+            amount = itemView.findViewById(R.id.debtAmount);
+            View.OnClickListener listener = new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    mSelectedDebt = getAdapterPosition();
+                    notifyItemRangeChanged(0, debtData.length);
```
Comment (Puh00) [inline app/src/main/java/com/goayo/debtify/view/adapter/PickDebtAdapter.java:92]: Very cool

File: app/src/main/java/com/goayo/debtify/view/adapter/UserCardViewAdapter.java [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+    /**
+     * Constructor for UserCardViewAdapter.
+     *
+     * @param dataSet Takes in a list of IUserData.
+     */
+    public UserCardViewAdapter(IUserData[] dataSet) {
+        this.dataSet = dataSet;
+    }
+
+    public UserCardViewAdapter(String username, String phoneNumber, View.OnClickListener clickListener) {
```
Comment (Puh00) [inline app/src/main/java/com/goayo/debtify/view/adapter/UserCardViewAdapter.java:43]: Add some generic Jdoc

**Review by OscarSanner — APPROVED** [review]
This looks good, as we implement the modelview package all the references to modelengine from view will be scrubbed.

**Review by Puh00 — APPROVED** [review]

File: app/src/main/java/com/goayo/debtify/view/SettleDebtFragment.java (hunk 6 above)
Comment (yenanw) [reply app/src/main/java/com/goayo/debtify/view/SettleDebtFragment.java:45]: will do in a new branch

**Comment by OlofSjogren** [comment]
My name is Olof and i approve of this message.

**Pull request merged**